packages/shared/src/hooks/useChannelFee.ts

```diff
@@ -0,0 +1,211 @@
+import { z } from "zod/v3";
+import { useDebounce } from "use-debounce";
+import { erc20Abi, Hex } from "viem";
+import { useEffect, useState } from "react";
+import { Decimal } from "decimal.js";
+import {
+  createEstimateChannelPostOrEditCommentFeeData,
+  estimateChannelEditCommentFee,
+  estimateChannelPostCommentFee,
+  TotalFeeEstimation,
+} from "@ecp.eth/sdk/channel-manager";
+import { PublicClient, Transport, Chain } from "viem";
+import { never } from "../helpers";
+import { EMPTY_PARENT_ID } from "@ecp.eth/sdk";
+
+export function useChannelFee(
```

Comment:
I wonder if we could use `useQuery()` instead of difficult `useEffect()` and all the logic around it, the `shared` package already has react-query as peer dependency.